modules/storage/main.tf

```diff
@@ -38,20 +38,26 @@ resource "google_storage_bucket" "bucket" {
   location                    = var.location
   force_destroy               = var.force_destroy
   uniform_bucket_level_access = var.uniform_bucket_level_access
+  labels                      = var.storage_bucket_labels
 
   versioning {
     enabled = var.versioning
   }
 
   dynamic "lifecycle_rule" {
-    for_each = var.expiration_days == null ? [] : [var.expiration_days]
+    for_each = var.lifecycle_rules
```

Comment:
This is breaking change and we will need an upgrade guide. Something similar to the diff you have in the readme should work.  Example https://github.com/terraform-google-modules/terraform-google-log-export/blob/master/docs/upgrading_to_v5.0.md